src/store/domains/ui.js

```diff
@@ -0,0 +1,19 @@
+
+import { makeAutoObservable } from 'mobx';
+import i18n from '~/i18n';
+
+export default class UI {
+  lang = 'en'
+
+  constructor() {
+    makeAutoObservable(this);
+  }
+
+
+  /* Actions */
+
+  setLanguage(lang) {
+    this.lang = lang;
+    i18n(lang);
```

Comment:
Die `i18n`-Funktion wurde zuvor in den Komponenten mit entsprechenden Sprachcodes aufgerufen.
Um den UI-Store aber in den Mittelpunkt zu verlegen, verschieben wir den `i18n`-Aufruf in den UI-Store und zwar wenn die Sprach im Store per Action gesetzt wird. So haben wir nur eine Stelle, wo `i18n` referenziert wird.